src/Entities/Entity.cpp

```diff
@@ -1293,7 +1293,7 @@ void cEntity::TickBurning(cChunk & a_Chunk)
 		m_TicksSinceLastFireDamage++;
 		if (m_TicksSinceLastFireDamage >= FIRE_TICKS_PER_DAMAGE)
 		{
-			if (!IsFireproof())
+			if (!(IsFireproof() || HasLava))
```

Comment:
I think the DeMorgan-ed version would be a bit easier to understand:
```cpp
if (!IsFireproof() && !HasLava)  // Not fireproof, and hasn't taken lava-damage already (#2197)
```

Reply:
Fixed.